.github/workflows/testing.yml

```diff
@@ -1,35 +1,51 @@
 name: Pytest MSS
 
 on: [ push, pull_request ]
+env:
+  PAT: ${{ secrets.PAT }}
 
 jobs:
   Test-MSS:
     runs-on: ubuntu-latest
-    strategy:
-      max-parallel: 5
+    
+    defaults:
+      run:
+        shell: bash
+
+    container:
+      image: yourmss/mss-test:latest
 
     steps:
     - uses: actions/checkout@v2
-
-    - name: Add conda to system path
-      run: |
-        # $CONDA is an environment variable pointing to the root of the miniconda directory, action
-        echo $CONDA/bin >> $GITHUB_PATH
-
-    - name: Setup Conda
+        
+    - name: Update dependencies if changed
       run: |
-        conda config --add channels conda-forge
-        conda config --add channels defaults
-        conda create -n mssenv python=3
+        cmp -s /meta.yaml localbuild/meta.yaml || (echo Dependencies differ \
+        && cat localbuild/meta.yaml \
+        | sed -n '/^requirements:/,/^test:/p' \
+        | sed -e "s/.*- //" \
+        | sed -e "s/menuinst.*//" \
+        | sed -e "s/.*://" > reqs.txt \
+        && conda install -n mssenv --file reqs.txt --force-reinstall)
```

Comment:
Mmh. This is nifty, but I am not sure I like it. In case of changed dependencies I'd expect us to necessarily and potentially manually update the docker image before we can proceed.
It is typically only necessary that current builds work within the image. At least for juwave we so far have only added stuff to it and try that the code works with both the current stable and unstable image.

Reply:
this enables to see that an ongoing build based on the meta.yaml rules does not give a working state. We need to do a fresh build to figure that there are newer packages with changed dependencies for us.

Reply:
I am also not sure I like it.

This is automating something that might be better off done by hand, yes.
However having something done automatically is always easier, and if it somehow breaks stuff, nothing happens to the stable images anyway.

Reply:
Again, I am not against merging this now. Let's how it works in practice.

Reply:
We are not alone with this idea.
https://github.com/conda/conda/issues/5126

When it makes trouble we can look if always build alpha is an option and to install also from that  --deps-only